From 829755af1277bad370f9f73416e3da44b243acb1 Mon Sep 17 00:00:00 2001
From: Yannick Serafini <yannick.serafini@wifx.net>
Date: Thu, 22 Jul 2021 12:15:23 +0200
Subject: [PATCH 01/13] Add optionnal customization of Atmel NAND PMECC
 parameters in dt PMECC bit can be defined with atmel,pmecc-cap PMECC sector
 size can be defined with atmel,pmecc-sector-size

Signed-off-by: Yannick Serafini <yannick.serafini@wifx.net>
---
 drivers/mtd/nand/raw/atmel/nand-controller.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/mtd/nand/raw/atmel/nand-controller.c b/drivers/mtd/nand/raw/atmel/nand-controller.c
index 3d13f22a205c..f7808f070470 100644
--- a/drivers/mtd/nand/raw/atmel/nand-controller.c
+++ b/drivers/mtd/nand/raw/atmel/nand-controller.c
@@ -209,6 +209,7 @@ struct atmel_nand_controller_ops {
 struct atmel_nand_controller_caps {
 	bool has_dma;
 	bool legacy_of_bindings;
+	bool allow_pmecc_params;
 	u32 ale_offs;
 	u32 cle_offs;
 	const char *ebi_csa_regmap_name;
@@ -1124,7 +1125,7 @@ static int atmel_nand_pmecc_init(struct nand_chip *chip)
 		return -ENOTSUPP;
 	}
 
-	if (nc->caps->legacy_of_bindings) {
+	if (nc->caps->legacy_of_bindings || nc->caps->allow_pmecc_params) {
 		u32 val;
 
 		if (!of_property_read_u32(nc->dev->of_node, "atmel,pmecc-cap",
@@ -2425,6 +2426,7 @@ static const struct atmel_nand_controller_caps atmel_sama5_nc_caps = {
 	.ale_offs = BIT(21),
 	.cle_offs = BIT(22),
 	.ops = &atmel_hsmc_nc_ops,
+	.allow_pmecc_params = true,
 };
 
 /* Only used to parse old bindings. */
-- 
2.43.0

